### PR TITLE
core/linux-aarch64: Update to current stable v6.1.13

### PR DESCRIPTION
**Pull request by lynix**
The current linux-aarch64 package version ships a kernel (v5.19.8) that is more than 4 months old.

The 5.19 branch is EOL as it's not an LTS version and there seem to be [multiple CVEs](https://www.linuxkernelcves.com/streams/5.19). This pull request updates to the current stable release.

Built (distcc, official cross-toolchain) and tested on an Odroid C4.

As requested by users I have made builds of this available in a [temporary repository](https://github.com/lynix/linux-aarch64-bin). They have been tested and confirmed to be working on multiple platforms, see [forum thread](https://archlinuxarm.org/forum/viewtopic.php?f=23&t=16280).

### CODE REVIEW

**Comment by ambiamber** [comment]
Hello to you,
Please set
CONFIG_PHY_MVEBU_CP110_UTMI=y
It is required for Mochabin USB, possibly other Marvell Armada systems as well.
See https://archlinuxarm.org/forum/viewtopic.php?f=65&t=16248 for more details.
Thank you,
Ambie

File: core/linux-aarch64/config
Comment (xdarklight) [inline core/linux-aarch64/config:9853]: This symbol was previously named `CONFIG_MESON_EFUSE` and marked as =m

File: core/linux-aarch64/config
Comment (xdarklight) [inline core/linux-aarch64/config:9858]: I think this symbol was called `CONFIG_ROCKCHIP_EFUSE`  before and marked as =y

**Review by xdarklight — CHANGES_REQUESTED** [review]

File: core/linux-aarch64/config
Comment (lynix) [reply core/linux-aarch64/config]: Thanks, fixed in e12d5b1.

File: core/linux-aarch64/config
Comment (lynix) [reply core/linux-aarch64/config]: Thanks, fixed in e12d5b1.

**Comment by lynix** [comment]
> Hello to you, Please set CONFIG_PHY_MVEBU_CP110_UTMI=y It is required for Mochabin USB, possibly other Marvell Armada systems as well. See https://archlinuxarm.org/forum/viewtopic.php?f=65&t=16248 for more details. Thank you, Ambie

Thanks, included as 5a6f3c0.

**Comment by ipha** [comment]
Thank you!

I have 6.1.10 is running successfully on an Ampere VM.

**Comment by ambiamber** [comment]
Hello @lynix 
Is there any chance you could put your PKGBUILD into AUR?  If you don't want to that's fine.
Regards,
Ambie

**Comment by ambiamber** [comment]
@kmihelich and @neteleven Are you there?

**Comment by ambiamber** [comment]
9 commits is 8 too many, see:
https://github.com/archlinuxarm/PKGBUILDs/pull/1976#issuecomment-1380187299

**Comment by lynix** [comment]
> 9 commits is 8 too many, see: [#1976 (comment)](https://github.com/archlinuxarm/PKGBUILDs/pull/1976#issuecomment-1380187299)

Thanks for reminding me of that one. Squashed.

**Comment by lynix** [comment]
Hi @Ambie,
 
I'm afraid this wouldn't be such a good idea as the package name would clash with the original from the [core] repo and this would likely break AUR auto-builds.
 
Maybe I could submit it under a different name, e.g. 'linux-aarch64-stable'. I'm already thinking about setting up a repository for the package.
 
But after all my real intent is still getting the maintainer to accept my PR.
 
 
Cheers,
 
lynix

**Comment by ambiamber** [comment]
You're welcome, however I still see there are two commits when they only want PRs with one commit.

**Comment by lynix** [comment]
> You're welcome, however I still see there are two commits when they only want PRs with one commit.

I know but I'm pretty sure 53aa9a8 can be seen highly controversial and I really want to give them the opportunity to drop it and merge the rest. I think two commits won't be too much to review, and [CONTRIBUTING](https://github.com/archlinuxarm/PKGBUILDs/blob/master/CONTRIBUTING.md) only says to squash incremental changes.

**Comment by ambiamber** [comment]
I would call it linux-aarch64-current

 

The AUR submission guildlines on the Arch wiki explains:

 

Use "conflicts" and "provides" not "replaces" to have a package effectivly change names.
 

Sent: Wednesday, February 08, 2023 at 2:23 AM

**Review by xdarklight — APPROVED** [review]

**Comment by lynix** [comment]
Well, v6.1.12 is out now and I'm really tempted to update this pull request. On the other hand that would mean it would need a new review again.

Any opinions on this? I'm starting to loose hope for this PR to be merged at all.

**Comment by lynix** [comment]
> [53aa9a8](https://github.com/archlinuxarm/PKGBUILDs/commit/53aa9a8e460afe79d4678dcef33cbabbfdc53567) doesn't look the right way (tm) to me. Do they just need a proper rebase on Linux 6.1?

In 53aa9a8 I just switched from applying patches via `git apply` to `patch -p1`. The former doesn't seem to work inside another repository so with this change I allow for building the package directly from the PKGBUILDs repository. I understand that this is just one way of building things and that ALARM maintainers may choose to not need it, so I left it separate.

What do you mean by 'proper rebase on Linux 6.1'? This change is not related to the kernel change at all.

**Comment by lategoodbye** [comment]
My assumption was that these 2 kernel patches doesn't apply via `git apply` because it's more strict (no fuzzyness) than `patch -p1`. Maybe you can post the result of git apply here. Why has PKGBUILD using `git apply` worked before?

**Comment by lynix** [comment]
> My assumption was that these 2 kernel patches doesn't apply via `git apply` because it's more strict (no fuzzyness) than `patch -p1`. Maybe you can post the result of git apply here. Why has PKGBUILD using `git apply` worked before?

Okay that's interesting. When I ran the version with `git apply` it didn't output anything but it didn't change anything either. So I assumed it's because I was building inside the PKGBUILD repository and that this would make the patches not apply to anything (as the kernel sources are untracked in the repository).

I don't know why it worked in the first place, and to be honest I really don't care. As I said in my very first post, I'd be happy if 53aa9a8 was just dropped as long as the rest would finally get merged (I'm actually starting to regret that I included it in the first place). Each day that is passing by is another day that someone is running a box with a kernel that is EOL and vulnerable.

@kmihelich: You obviously seem not too busy to push dozens of updates a day, for basically any package. Is it really too much to ask for _any_ kind of reaction to a pull request for a core component that has serious issues?

Any of the following would be appreciated:
- accept the pull request
- decline it
- tell me to F off because you don't like something about this PR
- tell people that _linux-aarch64_ is considered an unofficial kernel package and unsupported

Really, I'd be fine with any of the above. What I'm not fine with is wasting my time trying to contribute (i.e. do work for you or the project) with the aim of fixing the experience for existing ALARM users and being completely ignored.

Heck, it's been way easier to get patches accepted on LKML :)

**Comment by lynix** [comment]
So this is **v6.1.12**-1 with the controversal patch removed. I've changed my build setup and now the `git apply` method is working fine.

**Comment by xdarklight** [comment]
> @kmihelich: You obviously seem not too busy to push dozens of updates a day, for basically any package.

My understanding is that he has some automation that pushes updates released for Arch Linux (x86_64).

Still I fully agree with you: feedback from the Arch Linux ARM team on this topic would be great.
As stated before: `linux-aarch64` users are affected by several security issues.
In the past updates for `linux-aarch64` have also been on hold for a while (see #1835 and #1877).

**Comment by ipha** [comment]
Has anyone from @archlinuxarm been in contact with @kmihelich ? Are they still alive? Their last form post was over a year ago and they're not active on github.

**Comment by mhabb** [comment]
> So this is **v6.1.12**-1 with the controversal patch removed. I've changed my build setup and now the `git apply` method is working fine.

@lynix could you share what you did to make `git apply` work? Because it doesn't work for me.

**Comment by lynix** [comment]
> Has anyone from @archlinuxarm been in contact with @kmihelich ? Are they still alive? Their last form post was over a year ago and they're not active on github.

@kmihelich is [very active](https://github.com/archlinuxarm/PKGBUILDs/commits?author=kmihelich) (~5-15 commits a day).

**Comment by lynix** [comment]
> > So this is **v6.1.12**-1 with the controversal patch removed. I've changed my build setup and now the `git apply` method is working fine.
> 
> @lynix could you share what you did to make `git apply` work? Because it doesn't work for me.

Sure. What did the trick for me was to ensure that I'm not building from within the PKGBUILDs repository. So when setting up the clean chroot for building I'm bind-mounting just the package subfolder (core/linux-aarch64) into the chroot, not the entire repo.

My assumption for an explanation would be that this way `git` can't see that it's running inside a work tree and behaves like a one-shot replacement for `patch`. Haven't tested this theory, though.

**Comment by ambiamber** [comment]
I think there is the possibility that at least one of the issues here is that this kernel package will end up in the distribution media used to install ALARM. If something went wrong with the distribution media it would cascade. I think that having a separate package, linux-aarch64-current for current linux kernels would side-step that issue as linux-aarch64-current would not be used to install ALARM.

**Comment by ambiamber** [comment]
If trying to get linux-aarch64-current fails then two AUR packages linux-aarch64-current and linux-aarch64-current-bin could be created.

**Comment by lynix** [comment]
Having a vulnerable old kernel in the distribution media isn't good either, that's why this pull request is about upgrading it to a maintained release.

I'd really avoid having two separate kernel packages, that's just confusing to the user and doesn't solve the issue that existing installations would continue to run the kernel version that has security issues.

**Comment by microcai** [comment]
please include this patch as  Rock 4B actully use this SPI flash ship.
[0001-spi-nor-Add-support-for-xt25f32b-xt25f128b.patch.txt](https://github.com/archlinuxarm/PKGBUILDs/files/10782995/0001-spi-nor-Add-support-for-xt25f32b-xt25f128b.patch.txt)

**Comment by lynix** [comment]
> please include this patch as Rock 4B actully use this SPI flash ship. [0001-spi-nor-Add-support-for-xt25f32b-xt25f128b.patch.txt](https://github.com/archlinuxarm/PKGBUILDs/files/10782995/0001-spi-nor-Add-support-for-xt25f32b-xt25f128b.patch.txt)

Sorry, this PR is about upgrading the kernel package to a newer version. Adding a new custom patch is a different thing that would require a separate PR as we're not mixing unrelated changes (see [CONTRIBUTING](https://github.com/archlinuxarm/PKGBUILDs/blob/master/CONTRIBUTING.md)).

Or did this SPI flash chip work on v5.19.8 without the patch?

**Comment by xdarklight** [comment]
> please include this patch as Rock 4B actully use this SPI flash ship.
> 
> [spi-add-XT25F32.patch.txt](https://github.com/archlinuxarm/PKGBUILDs/files/10783085/spi-add-XT25F32.patch.txt)

This patch is missing at least a change in `drivers/mtd/spi-nor/core.c`.
If you want this to be included then please work with the upstream maintainers. There's been a recent discussion: [https://lore.kernel.org/lkml/20220201002535.2186484-1-dev.linux@crtified.me/T/]

**Comment by microcai** [comment]
> Or did this SPI flash chip work on v5.19.8 without the patch?

no.

**Comment by microcai** [comment]
> 

oh, yes. updated.

[0001-spi-nor-add-XT25F32-v2.patch.txt](https://github.com/archlinuxarm/PKGBUILDs/files/10784330/0001-spi-nor-add-XT25F32-v2.patch.txt)

**Comment by lynix** [comment]
> oh, forget it, now include core.{c,h} change as well.
> 
> [0001-spi-nor-add-XT25F32-v2.patch.txt](https://github.com/archlinuxarm/PKGBUILDs/files/10784333/0001-spi-nor-add-XT25F32-v2.patch.txt)

Again, please stop misusing this pull request for adding a new patch.

Open a separate pull request or (better) have a look at the LKML link for upstream integration status.

**Comment by fremouw** [comment]
@kmihelich what's holding you back from merging this? I think there are a lot of people waiting for this to be merged, you can't stick to an old not supported kernel. Thanks!

**Comment by microcai** [comment]
> Again, please stop misusing this pull request for adding a new patch.

just post here in case someone need it.

 
> Open a separate pull request or (better) have a look at the LKML link for upstream integration status.

need to wait this PR merged before I can open another PR.

**Comment by anarsoul** [comment]
Tested this PR with https://github.com/archlinuxarm/PKGBUILDs/pull/1968 on my Nanopi R5S, no issues so far.

**Comment by anarsoul** [comment]
@kmihelich I understand that you are busy with your daytime job and may not have enough time to review and merge it, but could you please delegate review of changes like this to some people who are involved in the project?

If you need more testing, I can volunteer to test it on my hardware - I have quite variety of aarch64 boards: rock64 with rk3328, rockpro64 with rk3399, nanopi r5s with rk3568, quartzpro64 with rk3588 (not supported yet in mainline), pine64-lts with allwinner a64.

Kernel upgrades are important, both security-wise and for supporting new hardware, so it would be nice to have regular kernel upgrades.

**Comment by danieletorelli** [comment]
@kmihelich, @graysky2 could you please help to move this forward or at least provide some constructive feedback on what could be improved to make it acceptable?

I understand the lack of time, but please realise that people here is offering to help you in many different ways and IMO the minimum you could do is being welcoming and inclusive and provide feedback rather than ignoring this help. Even more, if the help is offered for such important changes like this one.

**Comment by lynix** [comment]
Updated to **v6.1.13**.

With v6.2 in mainline I'm expecting to update this PR to 6.2.y soon.

As 6.1 has been promoted to an LTS branch I have created a new package [linux-aarch64-lts](https://aur.archlinux.org/packages/linux-aarch64-lts) in the AUR for those wanting to stick with an LTS kernel (note: this package does not support chromebooks as I've got no way of testing this).

**Comment by czka** [comment]
In addition to @anarsoul's list of hwardware:
> I can volunteer to test it on my hardware - I have quite variety of aarch64 boards: rock64 with rk3328, rockpro64 with rk3399, nanopi r5s with rk3568, quartzpro64 with rk3588 (not supported yet in mainline), pine64-lts with allwinner a64.

I can also test on MACCHIATObin Single Shot. I'm especially interested in bringing the USB 3.0 port functionality on that board back, which has been missing in Arm Arch Linux kernels for a year or 2 now. I guess [it's supposed to be fixed](https://github.com/archlinuxarm/PKGBUILDs/pull/1984/files#diff-e1e8b5358e7ab2d9e77acb3434b0005f053cd702ec3871edeb8c6b7dc6ca1607R9991) in this PR.

On weekend I'll try building this PR if I'm able to setup a [Distcc Cross-Compiling](https://archlinuxarm.org/wiki/Distcc_Cross-Compiling) with my x86 box. In case I don't, anybody knows maybe how long it might take to build kernel on a MACCHIATObin Single Shot with 4GB RAM, natively?

**Comment by lynix** [comment]
> On weekend I'll try building this PR if I'm able to setup a [Distcc Cross-Compiling](https://archlinuxarm.org/wiki/Distcc_Cross-Compiling) with my x86 box. In case I don't, anybody knows maybe how long it might take to build kernel on a MACCHIATObin Single Shot with 4GB RAM, natively?

I've set up a temporary repository containing pre-built packages from this PR in case you'd like to spare the build time and don't mind trusting pre-built packages. I don't know about the MACCHIATObin but my Odroid C4 took over 5 hours to build this natively. That's why I've switched to a Docker-based build environment using QEMU for running aarch64 binaries. That brings the build time down to about ~65min on a 48 core Threadripper.

**Comment by NextWork123** [comment]
@lynix We can consider update to 6.2.0-1 i already done the job for myself for my oracle cloud VM with ampere CPU but working to add some feature on it, as i'm working on cachyos. [repository](https://github.com/NextWork123/linux-aarch64-bore) Proof: 
immagine

**Comment by danieletorelli** [comment]
FYI: They released some hour ago `linux-aarch64-6.2.0-1` (https://github.com/archlinuxarm/PKGBUILDs/commit/5b197b6168e6f9365e5066508bf264987444f1b6).

**Comment by lynix** [comment]
Okay this is proof that @kmihelich is indeed alive and just unwilling to bless us with any reaction.

So thanks for nothing, I'll close this one as it's obsolete now. And a big thank you to everybody who has taken the time to test-drive my PR version (or pre-built packages), I appreciate it!

**Comment by fremouw** [comment]
heh, not sure if I should be happy that the kernel is updated or I should cry how @kmihelich is handling this. In my opinion this a good example of how to kill a community. Thanks for all your effort @lynix!

**Pull request closed without merging**
